subiquity/cloudinit.py

```diff
@@ -86,3 +138,31 @@ async def cloud_init_status_wait() -> (bool, Optional[str]):
     else:
         status = read_legacy_status(sp.stdout)
     return (True, status)
+
+
+async def validate_cloud_init_schema() -> tuple[bool, Optional[list[str]]]:
+    """Check for cloud-init schema errors.
+
+    Returns True if validated OK or can't tell. Otherwise, False and
+    a list of bad keys found.
+    Requires cloud-init supporting recoverable errors and extended status.
+    :return: tuple of (ok, list of bad keys or None)
+    """
+
+    if not supports_recoverable_errors():
+        return (True, None)
```

Comment:
I guess it would be more pythonic to have this raise with an exception with the list of bad keys, which could in turn be re-raised as that `AutoinstallError`.  What do you think?

Reply:
Yeah I think that makes sense. I'll rework the logic. 

Since that if section is highlighted: Part of my assumption here was that we needed recoverable error support to get the bad keys. i.e. we need to ask cloud-init if invalid cloud-config was something it encountered. I suppose this isn't necessarily true, and we could just query the schema always. This has the benefit of cutting out one of the subprocess calls. What do you think?

Reply:
That makes sense to me but let's VM confirm it with older images, where `supports_recoverable_errors() == False`